macro/e12014/curtis/viewer/run_eve_dQdtA.C

```diff
@@ -0,0 +1,146 @@
+   //TString evtInputDataFile = TString::Format("../unpacking/data/linked/evtRun_%04d.root", runNum);
+   std::cout << "Opening: " << evtInputDataFile << std::endl;
+   TString inputCuts = "/mnt/projects/hira/e12014/tpcSharedInfo/PIDcuts/Run323Yield.root"; // 150 torr
+
+   TString dir = getenv("VMCWORKDIR");
+   TString geoFile = "ATTPC_v1.1_geomanager.root";
+   TString mapFile = "e12014_pad_mapping.xml";
+   TString parameterFile = "ATTPC.e12014.par";
+
+   TString InputDataPath = InputDataFile;
+   TString OutputDataPath = OutputDataFile;
+   TString GeoDataPath = dir + "/geometry/" + geoFile;
+   TString mapDir = dir + "/scripts/" + mapFile;
+   TString digiParFile = dir + "/parameters/" + parameterFile;
+
+   FairRunAna *fRun = new FairRunAna();
+   FairRootFileSink *sink = new FairRootFileSink(OutputDataFile);
+   FairFileSource *source = new FairFileSource(InputDataFile);
+   fRun->SetSource(source);
+   fRun->SetSink(sink);
+   fRun->SetGeomFile(GeoDataPath);
+
+   FairRuntimeDb *rtdb = fRun->GetRuntimeDb();
+   FairParAsciiFileIo *parIo1 = new FairParAsciiFileIo();
+   //FairParRootFileIo *parIo1 = new FairParRootFileIo();
+   // parIo1->open("param.dummy.root");
+   //rtdb->setFirstInput(parIo1);
+
+   std::cout << "Setting par file: " << digiParFile << std::endl;
+   parIo1->open(digiParFile.Data(), "in");
+   rtdb->setFirstInput(parIo1);
+   rtdb->getContainer("AtDigiPar");
+
+   auto fMap = std::make_shared<AtTpcMap>();
+   fMap->ParseXMLMap(mapDir.Data());
+
+   AtViewerManager *eveMan = new AtViewerManager(fMap);
+   eveMan->SetCheckGood();
+   eveMan->SetCheckBranch("AtRawEventSub");
+
+   AtRawEvent *respAvgEvent;
+   TFile *f2 = new TFile("respAvg.root");
+   f2->GetObject("avgResp", respAvgEvent);
+   f2->Close();
+
+   auto threshold = 45;
+   auto filterOrder = 6;
+   auto cutoff = 75;
+   auto cycles = 4;
+
+   auto psa = std::make_unique<AtPSAIterDeconv>();
+   psa->SetResponse(*respAvgEvent);
+   psa->SetFilterOrder(filterOrder);
+   psa->SetCutoffFreq(cutoff);
+   psa->SetIterations(cycles);
+   // psa->SetThreshold(threshold);
+   auto sidePSA = new AtSidebarPSAIterDeconv(eveMan->GetSidebar());
+   sidePSA->SetPSA(psa.get());
+   AtPSAtask *psaTask = new AtPSAtask(std::move(psa));
+   psaTask->SetInputBranch("AtRawEventSub");
+   psaTask->SetOutputBranch("AtEventIter");
+   psaTask->SetPersistence(kTRUE);
+   eveMan->GetSidebar()->AddSidebarFrame(sidePSA);
+
+   AtCutHEIST *cut = new AtCutHEIST(evtInputDataFile, inputCuts);
+   cut->AddAllSpecies();
+   //cut->AddSpecies("Pb197");
+
+   auto boolMacEvt = std::make_unique<AtTabInfoFairRoot<AtRawEvent>>(eveMan->GetRawEventName());
+   AtBoolMacro *boolMac = new AtBoolMacro(TestTPC);
+   boolMac->GetTabInfo()->AddAugment(std::move(boolMacEvt));
+
+   AtDataReductionTask *reduceTask = new AtDataReductionTask();
+   reduceTask->SetInputBranch("AtRawEventSub");
+   reduceTask->SetReductionFunction([cut, boolMac]() { return (*cut)() && boolMac->ExecFunction(); });
```

Comment:
This is the bit I referenced in removing AtBoolMacro since I think it adds complexity without adding ability to do things.